sdk/storage/Azure.Storage.Common/src/Shared/Errors.Clients.cs

```diff
@@ -47,6 +47,10 @@ public static InvalidOperationException StreamMustBeAtPosition0()
         public static InvalidOperationException TokenCredentialsRequireHttps()
             => new InvalidOperationException("Use of token credentials requires HTTPS");
 
+        public static ArgumentException SasCredentialRequiresUriWithoutSas(Uri uri)
+            => new ArgumentException($"The resource URI {uri} must not contain SAS if AzureSasCredential is used." +
+                $" You can use BlobUriBuilder to strip SAS from the resource URI before instantiating clients.");
```

Comment:
What about something like?

```suggestion
            => new ArgumentException(
                $"You cannot use {nameof(AzureSasCredential)} when the resource URI also contains a Shared Access Signature: {uri}\n" +
                $"You can remove the shared access signature by creating a {nameof(BlobUriBuilder)}, setting {nameof(BlobUriBuilder.Sas)} to null, and calling {nameof(BlobUriBuilder.ToUri)}.");
```

Anyone getting this exception is probably going to be confused.  I'd like to put the URI after the core of the message so it's not lost behind a very long string.

Reply:
Sounds Good. I had to drop `nameof` for bloburibuilder as it doesn't exist in common.